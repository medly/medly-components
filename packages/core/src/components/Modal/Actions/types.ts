import { ScrollState } from '../types';

export type ModalActionUserProps = {
    /** Use this to align actions horizontally */
    alignItems?: 'left' | 'center' | 'right';
};

export interface StyledProps extends ModalActionUserProps {
    scrollState: ScrollState;
}
