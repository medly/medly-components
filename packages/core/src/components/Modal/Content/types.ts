import { ScrollState } from '../types';

export interface StyledProps {
    scrollState: ScrollState;
    headerHeight: number;
    overflowVisible?: boolean;
}
