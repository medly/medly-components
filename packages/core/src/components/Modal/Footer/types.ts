import { ScrollState } from '../types';

export type ModalFooterProps = {
    /** Use this to align actions horizontally */
    alignItems?: 'left' | 'center' | 'right';
};

export interface StyledProps extends ModalFooterProps {
    isSmallScreen: boolean;
    scrollState: ScrollState;
}
