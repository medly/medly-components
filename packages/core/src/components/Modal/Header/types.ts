import { HTMLProps, WithThemeProp } from '@medly-components/utils';
import { ModalProps, ScrollState } from '../types';

export interface Props extends ModalProps {
    setHeaderHeight?: (height: number) => void;
    isSmallScreen?: boolean;
}
export interface Props extends HTMLProps<HTMLDivElement>, WithThemeProp {
    id: string;
    scrollState: ScrollState;
}
