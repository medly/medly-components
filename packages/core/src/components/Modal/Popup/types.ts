import { HTMLProps } from '@medly-components/utils';

export interface ModalPopupProps extends HTMLProps<HTMLDivElement> {
    minWidth?: string;
    minHeight?: string;
    open?: boolean;
}
