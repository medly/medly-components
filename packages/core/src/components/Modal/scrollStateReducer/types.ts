interface ScrolledToTopAction {
    type: 'scrolledToTop';
    value: boolean;
}

interface ScrolledToBottomAction {
    type: 'scrolledToBottom';
    value: boolean;
}

interface ScrollPositionAction {
    type: 'scrollPosition';
    value: number;
}

export type ScrollActionTypes = ScrolledToTopAction | ScrolledToBottomAction | ScrollPositionAction;
