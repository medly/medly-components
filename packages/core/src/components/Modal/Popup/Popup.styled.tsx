import { breakpoints, media } from '@medly-components/utils';
import styled, { keyframes } from 'styled-components';
import { ModalPopupProps } from './types';

const mobileModalSlideIn = keyframes`
    0% {
        transform: translateY(100%);
    }
    100% {
        transform: translateY(0);
    }
`,
    mobileModalSlideOut = keyframes`
        0% {
            transform: translateY(0);
        }
        100% {
            transform: translateY(100%);
        }
`,
    desktopModalSlideIn = keyframes`
        0% {
            opacity: 0;
            transform: translateY(-2rem);
        }
        100% {
            opacity: 1;
            transform: translateY(0);
        }
`,
    desktopModalSlideOut = keyframes`
        0% {
            opacity: 1;
            transform: translateY(0);
        }
        100% {
            opacity: 0;
            transform: translateY(-2rem);
        }
`;

export const Popup = styled('div')<ModalPopupProps>`
    background: ${({ theme }) => theme.modal.backgroundColor};
    display: flex;
    flex-direction: column;
    position: relative;
    min-height: ${({ minHeight }) => minHeight || '25.6rem'};
    box-shadow: 0 0.4rem 3.2 ${({ theme }) => theme.modal.shadowColor};
    box-sizing: border-box;
    border-top-left-radius: 1.6rem;
    border-top-right-radius: 1.6rem;
    overflow: ${({ overflowVisible }) => !overflowVisible && `hidden`};

    ${({ theme, open }) => media(breakpoints(theme.breakpoints).down('S'))`
        width: 100%;
        height: calc(100% - 2.4rem);
        animation: ${open ? mobileModalSlideIn : mobileModalSlideOut} 0.4s cubic-bezier(0, 0, 0.33, 1);
    `}

    ${({ theme, minWidth }) => media(breakpoints(theme.breakpoints).between('M', 'L'))`
        border-radius: ${theme.modal.borderRadius};
        max-width: 72%;
        min-width: ${minWidth || '42%'};
        max-height: 80%;
    `}

    ${({ theme, minWidth }) => media(breakpoints(theme.breakpoints).up('XL'))`
        border-radius: ${theme.modal.borderRadius};
        min-width: ${minWidth || `60.5rem`};
        max-width: 103.6rem;
        max-height: 80%;
    `}

    ${({ theme, open }) => media(breakpoints(theme.breakpoints).up('M'))`
        animation: ${open ? desktopModalSlideIn : desktopModalSlideOut} 0.2s cubic-bezier(0, 0, 0.33, 1);
    `}
`;
