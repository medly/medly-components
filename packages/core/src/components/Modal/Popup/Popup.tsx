import { WithStyle } from '@medly-components/utils';
import type { FC } from 'react';
import { forwardRef, memo, Ref } from 'react';
import * as Styled from './Popup.styled';
import { ModalPopupProps } from './types';

const Component: FC<ModalPopupProps> = memo(forwardRef((props, ref: Ref<HTMLDivElement>) => <Styled.Popup ref={ref} {...props} />));

Component.displayName = 'Popup';
export const Popup: FC<ModalPopupProps> & WithStyle = Object.assign(Component, { Style: Styled.Popup });
