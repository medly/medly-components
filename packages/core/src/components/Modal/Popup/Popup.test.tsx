import { render } from '@test-utils';
import React from 'react';
import { Popup } from './Popup';
import { Props } from './types';

const ModalPopupRenderer = ({ open = true, isSmallScreen = true, minWidth, minHeight }: Props) => {
    return render(<Popup {...{ open, isSmallScreen, minWidth, minHeight }} />);
};

describe('Modal Popup component at small screen size', () => {
    it('should render properly when it is open', () => {
        const { container } = ModalPopupRenderer({ open: true, isSmallScreen: true, minWidth: '200px', minHeight: '200px' });
        expect(container).toMatchSnapshot();
    });

    it('should render properly when it is closed', () => {
        const { container } = ModalPopupRenderer({ open: false, isSmallScreen: true, minWidth: '200px', minHeight: '200px' });
        expect(container).toMatchSnapshot();
    });
});
