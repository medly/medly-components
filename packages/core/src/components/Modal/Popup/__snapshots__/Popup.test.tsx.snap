// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Popup component at small screen size should render properly when it is closed 1`] = `
.c0 {
  background: #ffffff;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-height: 200px;
  box-shadow: 0 0.4rem 3.2 rgba(67,84,101,0.5);
  box-sizing: border-box;
  border-top-left-radius: 1.6rem;
  border-top-right-radius: 1.6rem;
}

@media (max-width:767px) {
  .c0 {
    width: 100%;
    height: calc(100% - 2.4rem);
    -webkit-animation: buZueW 0.4s cubic-bezier(0,0,0.33,1);
    animation: buZueW 0.4s cubic-bezier(0,0,0.33,1);
  }
}

@media (min-width:768px) and (max-width:1439px) {
  .c0 {
    padding: 4.8rem 3.2rem;
    border-radius: 1.2rem;
    max-width: 72%;
    min-width: 200px;
    max-height: 80%;
  }
}

@media (min-width:1440px) {
  .c0 {
    padding: 4.8rem 3.2rem;
    border-radius: 1.2rem;
    min-width: 200px;
    max-width: 103.6rem;
    max-height: 80%;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-animation: dfPvjS 0.2s cubic-bezier(0,0,0.33,1);
    animation: dfPvjS 0.2s cubic-bezier(0,0,0.33,1);
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Modal Popup component at small screen size should render properly when it is open 1`] = `
.c0 {
  background: #ffffff;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-height: 200px;
  box-shadow: 0 0.4rem 3.2 rgba(67,84,101,0.5);
  box-sizing: border-box;
  border-top-left-radius: 1.6rem;
  border-top-right-radius: 1.6rem;
}

@media (max-width:767px) {
  .c0 {
    width: 100%;
    height: calc(100% - 2.4rem);
    -webkit-animation: cQwfQi 0.4s cubic-bezier(0,0,0.33,1);
    animation: cQwfQi 0.4s cubic-bezier(0,0,0.33,1);
  }
}

@media (min-width:768px) and (max-width:1439px) {
  .c0 {
    padding: 4.8rem 3.2rem;
    border-radius: 1.2rem;
    max-width: 72%;
    min-width: 200px;
    max-height: 80%;
  }
}

@media (min-width:1440px) {
  .c0 {
    padding: 4.8rem 3.2rem;
    border-radius: 1.2rem;
    min-width: 200px;
    max-width: 103.6rem;
    max-height: 80%;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-animation: hvOvm 0.2s cubic-bezier(0,0,0.33,1);
    animation: hvOvm 0.2s cubic-bezier(0,0,0.33,1);
  }
}

<div>
  <div
    class="c0"
    open=""
  />
</div>
`;
